sphinx/addnodes.py

```diff
@@ -200,6 +200,7 @@ class index(nodes.Invisible, nodes.Inline, nodes.TextElement):
     """
 
 
+# TODO(stephenfin): Remove this in Sphinx 2.0
```

Comment:
I actually tried to add `__init__` here so we could raise a warning, but it seems we have a `Centered` directive which needs to be removed first.

Reply:
Why removing the directive?

Reply:
See the comment - this is deprecated since before 2010

Reply:
This node is surely marked as deprecated. But `centered` directive is not marked as deprecated.

IMO, the directive is not important for sphinx-core. so +1 for deprecation. But it would be nice if we can split it into other package (for workaround).